File: Code TranslaterTests/Scripts/Python/9/9.cs
class Solution
{
	List<Object> ans = new List<object>();
	
	List<int> getLonelyNodes(TreeNode? root)
	{
		void helper(object root)
		{
			if (root.left is null && root.right)
			{
				this.ans.append(root.right.val);
				return helper(root.right);
			}
			if (root.left && root.right is null)
			{
				this.ans.append(root.left.val);
				return helper(root.left);
			}
			if (root.left && root.right)
			{
				return new List<object> { helper(root.left), helper(root.right) };
			}
			if (root.left is null && root.right is null)
			{
				return;
			}
		}
		
		helper(root);
		ans = this.ans;
		this.ans = new List<object>();
		return ans;
	}
}

File: Code TranslaterTests/Scripts/Python/9/9.py
class Solution:
    ans = []

    def getLonelyNodes(self, root: Optional[TreeNode]) -> List[int]:
        def helper(root):
            if root.left is None and root.right:
                self.ans.append(root.right.val)
                return helper(root.right)
            if root.left and root.right is None:
                self.ans.append(root.left.val)
                return helper(root.left)
            if root.left and root.right:
                return helper(root.left), helper(root.right)
            if root.left is None and root.right is None:
                return

        helper(root)
        ans = self.ans
        self.ans = []
        return ans